sdk/core/perf/src/random_stream.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/platform.hpp>
+
+#include "azure/perf/random_stream.hpp"
+
+#if defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+#endif
+
+static thread_local std::mt19937_64 random_generator(std::random_device{}());
+
+namespace {
+
+static uint8_t RandomChar()
+{
+  const uint8_t charset[] = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  std::uniform_int_distribution<size_t> distribution(0, sizeof(charset) - 2);
+  return charset[distribution(random_generator)];
+}
+
+void RandomBuffer(uint8_t* buffer, size_t length)
+{
+  uint8_t* start_addr = buffer;
+  uint8_t* end_addr = buffer + length;
+
+  const size_t rand_int_size = sizeof(uint64_t);
+  while (uintptr_t(start_addr) % rand_int_size != 0 && start_addr < end_addr)
```

Comment:
What is this loop condition doing? Why `mod rand_int_size ! = 0`? Add a comment.

Reply:
lol, I stole this code from @Jinming-Hu  hahhaha, He provided this cool code to fill a buffer with random data.

The loop is going from a start_addr to another memory addr, increasing the addrs on every loop and overriding the content.  It also contains what I would call a `overflow-detection` to break the loop if the ptr is overflow and comes back to 0 (as it is unsigned).  If the addr gets to the same size as `uint64_t`, the modulo operation will return 0 and break the loop.

I don't think we need a comment. The overflow detection is kind of common